internal/controllers/__snapshots__/instance_controller_test.snap

```diff
@@ -426,13 +426,6 @@ SnapShot = """
   },
   \"spec\": {
     \"ports\": [
-      {
-        \"name\": \"main\",
-        \"protocol\": \"TCP\",
-        \"port\": 80,
-        \"targetPort\": 80,
-        \"nodePort\": 30000
-      },
```

Comment:
Yes.
We create a netrule from the template svc and ingress, and generate the network information for svc and ingress from that netrule.
Therefore, if there is a mismatch between the template and the regenerated one, both will be reflected in svc and ingress.
To prevent this, only the regenerated information from the netrule is reflected.
The current test code doesn't convey that intent, so it should be fixed.

Reply:
Currently, The idea of Workspace Network Rule is an abstraction of Instance Network Override spec. And the idea of Instance Network Override spec is to update or insert predefined Ingress Rules and Service Ports on Template. Override processes in Instance reconciliation.

It does not matter whatever the before Instance Override Spec. New Rules and Ports are generated from Instance Network Override spec everytime in Instance reconciliation and they are applied to predefined Rules and Ports in Template. 

It comes from because if Overrides replaced all Rules and Ports in predefined in Template, we have to set the redundant Rules and Ports in InstanceOverrideSpec on every Instance,  which are the same with the predefined in Template.

Issue #668 ports duplication occured because whether update or insert is decided by Service Port "name". Currently we have to recognize the Service main port so there are Workspace Service Main Port in WorkspaceConfig and its annotation on Workspace Template.

If changing the idea, we need to rethink about the idea of WorkspaceConfig and InstanceNetworkOverrideSpec and so on.